Unmarshal array attributes not supported?

Hi,

I'm trying to unmarshal this json
```json
{
  "data": [
    {
      "id": "87233c69-cf45-401c-aa73-7e65b4eda96a",
      "type": "images",
      "attributes": {
        "app_id": "fac73220-881a-4a32-bc18-bdaa71666a26",
        "image_name": "nginx:latest",
        "cmd": null,
        "is_running": false,
        "instances": 6,
        "mem": 512,
        "envs": null,
        "port_mappings": null,
        "ports": [
          {
            "protocol": "tcp",
            "number": "80"
          }
        ]
      },
      "relationships": {
        "app": {
          "data": {
            "type": "apps",
            "id": "fac73220-881a-4a32-bc18-bdaa71666a26"
          }
        }
      }
    }
  ]
}
```

The `ports` attribute is an array data.
So I wrote a code like this

```go
type Image struct {
	Envs         []map[string]interface{} `jsonapi:"name=envs"`
	Ports        []map[string]interface{} `jsonapi:"name=ports"`
	PortMappings []map[string]interface{} `jsonapi:"name=port_mappings"`
	ID        string
	ImageName string    `jsonapi:"name=image_name"`
	CreatedAt time.Time `jsonapi:"name=created_at"`
	UpdatedAt time.Time `jsonapi:"name=updated_at"`
	App       *App      `jsonapi:"-"`
	Mem       int       `jsonapi:"name=mem"`
	AppId     string    `jsonapi:"name=app_id"`
	Instances int       `jsonapi:"name=instances"`
	isRunning bool      `jsonapi:"name=is_running"`
	Cmd       string    `jsonapi:"name=cmd"`
}

func (i Image) GetID() string {
	return string(i.ID)
}

func (i *Image) SetID(ID string) error {
	i.ID = ID
	return nil
}

func (i *Image) SetToOneReferenceID(name, ID string) error {
	if name == "app" {
		if ID == "" {
			i.App = nil
		} else {
			i.App = &App{ID: ID}
		}

		return nil
	}

	return errors.New("There is no to-one relationship with the name " + name)
}
```

But it panics.

```
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method

goroutine 1 [running]:
reflect.valueInterface(0x2e21c0, 0xc8203e80c0, 0xe1, 0x1, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/reflect/value.go:912 +0xe7
reflect.Value.Interface(0x2e21c0, 0xc8203e80c0, 0xe1, 0x0, 0x0)
	/usr/local/Cellar/go/1.5.1/libexec/src/reflect/value.go:901 +0x48
github.com/manyminds/api2go/jsonapi.UnmarshalInto(0xc8204842d0, 0x11a4080, 0x3cc280, 0xc82052ce38, 0x0, 0x0)
	project/vendor/src/github.com/manyminds/api2go/jsonapi/unmarshal.go:272 +0x1faf
github.com/manyminds/api2go/jsonapi.Unmarshal(0xc8204842d0, 0x2c5800, 0xc820129ae0, 0x0, 0x0)
```

I'm sorry if I'm doing something wrong or missing reading the docs again but could you tell me how to fix this?

--- a/jsonapi/fixtures_test.go
+++ b/jsonapi/fixtures_test.go
@@ -457,3 +457,22 @@ func (e ErrorRelationshipPosts) SetToOneReferenceID(name, ID string) error {
 func (e ErrorRelationshipPosts) SetToManyReferenceIDs(name string, IDs []string) error {
 	return errors.New("this also never works")
 }
+
+type Image struct {
+	ID    string      `json:"-"`
+	Ports []ImagePort `json:"image-ports"`
+}
+
+func (i Image) GetID() string {
+	return i.ID
+}
+
+func (i *Image) SetID(ID string) error {
+	i.ID = ID
+	return nil
+}
+
+type ImagePort struct {
+	Protocol string `json:"protocol"`
+	Number   int    `json:"number"`
+}

--- a/jsonapi/unmarshal_test.go
+++ b/jsonapi/unmarshal_test.go
@@ -72,6 +72,38 @@ var _ = Describe("Unmarshal", func() {
 			Expect(posts).To(Equal([]SimplePost{firstPost, secondPost}))
 		})
 
+		It("unmarshals array attributes into array structs", func() {
+			expected := Image{
+				ID: "one",
+				Ports: []ImagePort{
+					{
+						Protocol: "something",
+						Number:   666,
+					},
+				},
+			}
+
+			var actual Image
+			err := Unmarshal([]byte(`
+			{
+				"data": {
+					"type": "images",
+					"id": "one",
+					"attributes": {
+						"image-ports": [
+						{
+							"protocol": "something",
+							"number": 666
+						}
+						]
+					}
+				}
+			}
+      `), &actual)
+			Expect(err).To(BeNil())
+			Expect(actual).To(Equal(expected))
+		})
+
 		It("errors on existing record that does not implement MarshalIdentifier", func() {
 			type invalid struct{}
 			invalids := []interface{}{invalid{}}

